Xamarin.Forms.Platform.Android/Extensions/EntryRendererExtensions.cs

```diff
@@ -4,6 +4,28 @@ namespace Xamarin.Forms.Platform.Android
 {
 	internal static class EntryRendererExtensions
 	{
+
+		internal static ImeAction ToAndroidImeAction(this ReturnType returnType)
+		{
+			switch (returnType)
+			{
+				case ReturnType.Go:
+					return ImeAction.Go;
+				case ReturnType.Next:
+					return ImeAction.Next;
+				case ReturnType.Send:
+					return ImeAction.Send;
+				case ReturnType.Search:
+					return ImeAction.Search;
+				case ReturnType.Done:
+					return ImeAction.Done;
+				case ReturnType.Default:
+					return ImeAction.Done;
+				default:
+					throw new System.Exception("Return Type Not Supported");
```

Comment:
Maybe `ArgumentOutOfRangeException` or `NotImplemented` or something more specific, and include the value of the unsupported `ReturnType` in the error message.